test/truncate.c

```diff
@@ -19,6 +19,44 @@
 #define ONE_GIG_SIZE ((loff_t)1024 * 1024 * 1024)
 #define HALF_GIG_SIZE ((loff_t)512 * 1024 * 1024)
 
+static int test_truncate(struct io_uring *ring, int fd)
+{
+	struct io_uring_cqe *cqe;
+	struct io_uring_sqe *sqe;
+	int ret;
+
+	sqe = io_uring_get_sqe(ring);
+	if (!sqe) {
+		fprintf(stderr, "get sqe failed\n");
+		goto err;
+	}
+
+	memset(sqe, 0, sizeof(*sqe));
+
+	io_uring_prep_rw(IORING_OP_FTRUNCATE, sqe, fd, "fail", 0, 4);
+
+	ret = io_uring_submit(ring);
+	if (ret <= 0) {
+		fprintf(stderr, "sqe submit failed: %d\n", ret);
+		goto err;
+	}
+
+	ret = io_uring_wait_cqe(ring, &cqe);
+	if (ret < 0) {
+		fprintf(stderr, "wait completion %d\n", ret);
+		goto err;
+	}
+	ret = cqe->res;
+	io_uring_cqe_seen(ring, cqe);
+	if (cqe->res != -EINVAL) {
```

Comment:
Doesn't matter too much here, but you cannot use the cqe after calling io_uring_cqe_seen() as it could've been overwritten by a new cqe. You assign 'ret' here anyway, so please just use ret. Would be a shame if people copied an error like that for their app.

Reply:
Fixed!